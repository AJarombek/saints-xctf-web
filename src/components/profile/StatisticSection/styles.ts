/**
 * JSS styles for the StatisticsSection component.
 * @author Andrew Jarombek
 * @since 11/14/2020
 */

export default {
  stats: {},
  section: {}
};
