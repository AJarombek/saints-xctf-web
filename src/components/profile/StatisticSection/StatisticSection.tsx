/**
 * Component for displaying a list of statistics.
 * @author Andrew Jarombek
 * @since 11/14/2020
 */

import React, { ReactNode } from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';

interface Props {
  title: ReactNode;
  sections: Array<object>;
}

const useStyles = createUseStyles(styles);

const StatisticSection: React.FunctionComponent<Props> = ({ title, stats }) => {
  const classes = useStyles();

  return (
    <div className={classes.stats}>
      <h5>{title}</h5>
      {stats.map((stat, i) => (
        <div key={i} className={classes.section}>
          <p>{stat.name}</p>
          <p>{stat.value}</p>
        </div>
      ))}
    </div>
  );
};

export default StatisticSection;
