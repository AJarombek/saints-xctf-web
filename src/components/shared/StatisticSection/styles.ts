/**
 * JSS styles for the StatisticsSection component.
 * @author Andrew Jarombek
 * @since 11/14/2020
 */

import { FontMixins } from '../../../styles/mixins';

export default {
  stats: {
    margin: '20px 25px',

    '& > h3': {
      ...FontMixins.robotoBold(),
      marginBottom: 10,
    },
  },
  section: {
    display: 'flex',

    '& > p': {
      margin: '4px 0',
    },

    '& > p:nth-child(1)': {
      ...FontMixins.robotoSlab(),
      width: 200,
    },

    '& > p:nth-child(2)': {
      ...FontMixins.robotoThin(),
      width: 125,
      textAlign: 'end',
    },
  },
  '@media screen and (max-width: 450px)': {
    stats: {
      '& > h3': {
        fontSize: 16,
      },
    },
    section: {
      '& > p': {
        fontSize: 14,
      },

      '& > p:nth-child(1)': {
        width: 160,
      },

      '& > p:nth-child(2)': {
        width: 100,
      },
    },
  },
};
