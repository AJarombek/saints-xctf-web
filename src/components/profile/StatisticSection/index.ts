import StatisticSection from './StatisticSection';

export default StatisticSection;
