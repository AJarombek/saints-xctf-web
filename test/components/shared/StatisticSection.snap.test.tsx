/**
 * Snapshot test for the StatisticSection component.
 * @author Andrew Jarombek
 * @since 3/8/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import StatisticSection from '../../../src/components/shared/StatisticSection/StatisticSection';
import { milesStats } from '../../test-utils/statsMocks';

describe('StatisticSection Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <StatisticSection stats={milesStats} title="Statistic Section Snapshot" />
        </>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
