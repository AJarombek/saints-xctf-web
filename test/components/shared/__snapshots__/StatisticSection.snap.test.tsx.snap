// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticSection Snapshot Tests renders correctly 1`] = `
<div
  className="stats-0-1-1"
>
  <h3>
    Statistic Section Snapshot
  </h3>
  <div
    className="section-0-1-2"
  >
    <p>
      Miles All Time
    </p>
    <p>
      50
    </p>
  </div>
  <div
    className="section-0-1-2"
  >
    <p>
      Miles Past Year
    </p>
    <p>
      40
    </p>
  </div>
  <div
    className="section-0-1-2"
  >
    <p>
      Miles Past Month
    </p>
    <p>
      30
    </p>
  </div>
  <div
    className="section-0-1-2"
  >
    <p>
      Miles Past Week
    </p>
    <p>
      20
    </p>
  </div>
</div>
`;
